app.rb

```diff
@@ -0,0 +1,146 @@
+require_relative './student'
+require_relative './book'
+require_relative './teacher'
+require_relative './rental'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    @books.each { |b| puts "Title: #{b.title}, Author: #{b.author}" }
+  end
+
+  def list_all_people
+    @people.each { |p| puts "[#{p.class}] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}" }
+  end
+
+  def parent_permission?
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.capitalize
+    case parent_permission
+    when 'Y'
+      parent_permission = true
+    when 'N'
+      parent_permission = false
+    else
+      puts 'Please enter a valid input'
+      parent_permission?
+    end
+
+    parent_permission
+  end
+
+  def person_choice
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    answer = gets.chomp.to_i
+    if [1, 2].include?(answer)
+      answer
+    else
+      puts 'Please enter a valid value (1 or 2)'
+      person_choice
+    end
+  end
+
+  def create_a_person
+    answer = person_choice
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+
+    case answer
+    when 1
+      parent_permission = parent_permission?
+      @people << Student.new(age, name, parent_permission)
+    when 2
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(specialization, age, name)
+    end
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] Kindly fix the following error when I try to add parent permission or add specialization for a teacher I got it. In addition, make sure to create rental work correctly because I can't create rental without a person created.
>![image](https://user-images.githubusercontent.com/53402600/193094237-69f73753-c26d-423d-b49c-0e8caf9a6818.png)

Reply:
it has been fixed. Thank you